instrumentation/opentelemetry-instrumentation-asgi/src/opentelemetry/instrumentation/asgi/__init__.py

```diff
@@ -354,9 +354,12 @@ def collect_request_attributes(
             result, path, path, query_string, sem_conv_opt_in_mode
         )
     if http_url:
-        _set_http_url(
-            result, remove_url_credentials(http_url), sem_conv_opt_in_mode
-        )
+        if _report_old(sem_conv_opt_in_mode) or (
```

Comment:
I think _report_old already covers the cases for default and dup, no?

Reply:
_report_old cannot cover some _both_semconv test cases.

Reply:
I think we only want this for old semconv. In other words, no `url.full` in new/both semconv

Reply:
Thanks so much for your insight! I tested changes locally and will submit it very soon.

Reply:
Marking as draft

Reply:
Updated PR and test with instrumentations I commented. No url.full in new and both semconv with the change.